Adds request-direct-access-url file name option

canpan14 opened the pull request:
In support of this pull request https://github.com/Alfresco/alfresco-community-repo/pull/2065

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Alfresco/rest-api-explorer?pullRequest=197) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Alfresco/rest-api-explorer?pullRequest=197) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Alfresco/rest-api-explorer?pullRequest=197) it.</sub>

tpage-alfresco commented on src/main/webapp/definitions/alfresco-core.yaml:
```diff
@@ -10683,6 +10701,9 @@ definitions:
       attachment:
         type: boolean
         description: URL type (embedded/attachment).
+      fileName:
```
I think we'd usually have the items from the "Create" entity in the response too - i.e. `DirectAccessUrl` below. I don't think this is explicitly called out in the v1 API guidelines though.

I've added our architect team to review too, since they usually review additions to the v1 API.

canpan14 replied:
I could add it to the `DirectAccessUrl` class in alfresco-community-repo, but it looks like the current object structure is fed in from the content store itself. Not exactly sure where that code is or how dangerous it is to alter.
I could make `DirectAccessUrlResponse` class for the purposes of extending an adding on the additional field.
It's not hard to return the field, I just want to keep things in line with your expectations for the code base as a whole.

jamalkm replied:
@tpage-alfresco is correct, please make sure to add the file name to the response too. 
In terms of implementation, you have already modified the `ContentServiceImpl` class, so **adding** an attribute to `DirectAccessUrl` (part of our Java public API) is not a problem.

canpan14 replied:
@jamalkm The issue is the call to the content store serialized directly into` DirectAccessUrl`. While the serialization might still work with an extra field added, I worry it might start to diverge from representing the data object actually returned from the database vs. what we are returning from the API.

```
DirectAccessUrl directAccessUrl = null;
if (store.isContentDirectUrlEnabled())
{
    try
    {
        // referencing this
        directAccessUrl = store.requestContentDirectUrl(contentUrl, attachment, fileName, contentMimetype, validFor);
    }
    catch (UnsupportedOperationException ex)
    {
        // expected exception
    }
}
return directAccessUrl;
```
I can still open a PR on it though

canpan14 replied:
https://github.com/Alfresco/alfresco-community-repo/pull/2081

jamalkm reviewed (changes requested):
The PR looks good, just filename needs to be added to the response too.

tpage-alfresco commented:
Argh - sorry. I thought this had already been approved. I merged the main PR earlier today. Shall I revert the main PR back to a branch, or is everyone happy to update the response in a second PR?

jamalkm commented:
> Argh - sorry. I thought this had already been approved. I merged the main PR earlier today. Shall I revert the main PR back to a branch, or is everyone happy to update the response in a second PR?

I'm in favour of a second PR.

canpan14 commented:
Second PR to add file name to the response object https://github.com/Alfresco/alfresco-community-repo/pull/2081

canpan14 commented:
@jamalkm I added fileName to the response object in the api specs and noticed that `expiryTime` was improperly listed in the response object as 'expiresAt` so I fixed that as well.

jamalkm commented on src/main/webapp/definitions/alfresco-core.yaml:
```diff
@@ -10711,10 +10732,13 @@ definitions:
         type: boolean
         description: Flag to control the download method, **true** for attachment URL, **false** for embedded URL
-      expiresAt:
+      expiryTime:
```
Please revert the field to `expiresAt`. The name is based on our RestAPI guidelines. Ideally we should fix the attribute name in the `DirectAccessUrl` class but since it is part of our Java Public APIs we can't just rename it...

canpan14 replied:
Done!
